### ISSUE TITLE
Compensated support needed.

### ISSUE THREAD

**Issue by flow28** [clach04/pebble-send-message#1]
Add two lines of code 

Please contact me. 

Keep up the great work!

**Comment by flow28** [comment]
Hey,
Greetings from Berlin, Germany.
Are you by any chance the developer of the great Send Message Pebble app? 
If your answer is yes then the following message is for you: 

First I wanted to let you know that there are people out there who really love your work and make use of it daily. 
Great Work! 
I especially love your simple yet elegant Send Message App for Pebble, which I need several times a day e.g. to control my home.
However, there are two tiny things lacking, which would otherwise make it perfect for me.
I know it's been a long while since you worked on this project, but both requests need only are very tiny , one need the removal of one line of code and one the addition of one line.

Could you help me? 
It's really simple but I'm surely lacking the skills and tools. 

I'm looking forward to your reply. 

Cheers, 
Florian

**Comment by clach04** [comment]
Hallo Florian!

I'm not the original author (that would be Peter Summers) see the project description for links. He posted the source code so I spent some time wrestling it into a Git repo (and onto GitHub).

If you have a code change and want to make pull-request (PR), please go ahead and fork, make the change and submit a PR so I can take a look.

I recently posted to my watchframe work repo a way to get builds built using GitHub actions without needing build tools on your machine. Take a look at https://github.com/clach04/pebble_watchface_framework/tree/master/.github/workflows

I'm curious what your changes are :-)

Warm regards,

Chris

**Comment by clach04** [comment]
I just set up https://github.com/clach04/pebble-send-message/tree/gha_test - which auto builds on a push to that branch, check out https://github.com/clach04/pebble-send-message/actions (click a build, then scroll the bottom to locate a PBW to  download) 😺 .

If that works for you I can move it into master 🤞

**Comment by clach04** [comment]
@flow28 I just updated the master branch with an auto builder so let me know what changes you are thinking of and we can see if we can get your idea implemented 😊

**Comment by flow28** [comment]
Hey Chris, 
Thanks for the prompt reply and sorry for my late one. It's not that I'm lacking interest it's just that I have a newborn son since a few days. 
Surely you are able to help me with my problem. 
Just to put you into the picture: 
My wife and I are still rocking Pebbles and hope to continue doing so as long as we are able to sideload the pebble app to eventually appearing new iPhones and the connection isn't severed with an OS update in the future. 
We use the three configurable buttons in the Send Message Pebble App to do the following: 
I Open our Nuki Doorlock 
II Open our Driveway 
III Open our Garage 

Since we do this several times a day it's our most frequently use app by far. 
Due to the design of the app there are some minor inconveniences involved which in turn lead to somewhat of a safety hazzard. 
This leads to my wishes, which are quite simple actually. 
When pushing a button in the app and triggering an action an overlay window appears, which confirms the trigger but only inside of the app (confirm that button was pressed), but not from the receiver side. This notification is quite pointless for us. 
I'd like to disable it. So the line of the code which pushes the prompt when an action is triggered just can be deleted, or even better: make it appear only for 3-5 seconds and then have it dissaper automattically. 
So far you have to confirm it manually. This adds several mandatory button presses to carry out our changing routines. 
As we sometimes (but not every time) trigger all three actions after each other (but in different orders when arriving or leaving) this leads to a series of long button presses of up to eight times to start app, trigger (x3), confirm (x3) exit app. We don't have time for button smashing when driving the car or carrying groceries. 
Thus, we look to either have the notification disabled completely (delete line) or have it vanish after the run out of a timer (add auto close/confirm after 3-5s). 
Vibrate on triggering an action to notify you would still be fine. 

The second thing is the necessity of an auto-close of the app. 
We often forget to press the back button a final time on the watch which is needed to close the app and it stays on the watch screen listening all the time. Thus, a single accidential button press during the day can trigger an action to instantly unlocks our front door when we are not home, which actually happened several times already. 
Thus a new line of code to automatically close the app and return to the watchface after two minutes of inactivity would be cool. 
And that's basically it. 

TLDR without explanantion: 
Remove line of code for confirmation popup or modify line for it to auto close after 5s. 
Add line of code to close app after two minutes of inactivity. 

These simple changes make a tremendous difference for us and we are very interested to have them incorporated into the app. 
Unfortunately we lack even the tiniest ability to write code and are unable to do it ourself. 
I registered to GitHub specifically to get into contact with you to get some help. 
Can you help me? 
It would be highly appreciated and honored in your preferred way. 

I hope for the best. 
Cheers
Florian

**Comment by clach04** [comment]
@flow28 congratulations to you both!

I've opened up issues for each request:

  * #3 app exit timeout
  * #4 notification timeout
  * #5 don;t display notifications for success

None of these are simple one line code changes. The pop-up code is multiple notifications (some so quick you might not see them).

I've implemented a hard code hack for #3 based on code I wrote for another Pebble app, take a look and see if that helps. It may get you by rather than being a complete solution (it needs more work, for example config).

You might want to take a look at the other Pebble apps that call servers, they are more complex to set up though so you may be better off with send-message. I've not had chance to update the readme with links to al the alternatives I'm aware of.

**Comment by clach04** [comment]
@flow28 test implementations for all the above are now available, took me about a day. #5 has all 3 enhancements, all hard coded to be enabled. NO CONFIG.

Test build for #4 is missing #5, etc.

Hopefully they work or you :-)
